src/qibojit/tests/test_gates.py

```diff
@@ -31,8 +31,8 @@
     ],
 )
 def test_apply_gate(backend, nqubits, target, controls, dtype):
-    state = random_state(nqubits, dtype=dtype)
-    matrix = random_unitary(1, dtype=dtype)
+    state = random_statevector(2**nqubits)
+    matrix = random_unitary(2**1)
```

Comment:
I would prefer to keep the dtype argument here and in all other tests to make sure that both types work. For example for cupy the compiled kernels are different for each type, so if one works does not necessarily mean that the other one works too.

Reply:
The original functions don't have a `dtype` argument and are not backend agnostic. However, I can use `cast` to solve both issues.